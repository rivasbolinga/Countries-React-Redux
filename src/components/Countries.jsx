import React from 'react';
import styled from 'styled-components';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

const Countries = ({
  num, flag, name, capital, population, region,
}) => (
  <Wrapper className="country-card">
    <Link to={`/country/${num}`} className="link">
      <img src={flag.png} alt={flag.alt} className="flag" />
      <h2>{name}</h2>
      <p>
        <span>Capital:</span>
        {' '}
        {capital}
      </p>
      <p>
        <span>Population:</span>
        {' '}
        {population}
      </p>
      <p>
        <span>Region:</span>
        {' '}
        {region}
      </p>
    </Link>
  </Wrapper>
);

Countries.propTypes = {
  num: PropTypes.number.isRequired,
  flag: PropTypes.shape({
    png: PropTypes.string.isRequired,
    alt: PropTypes.string.isRequired,
  }).isRequired,
  name: PropTypes.string.isRequired,
  capital: PropTypes.string.isRequired,
  population: PropTypes.number.isRequired,
  region: PropTypes.string.isRequired,
};

const Wrapper = styled.section`
  border: 1px solid;
  .flag {
    width: 200px;
    height: auto;
  }
}

`;

export default Countries;
