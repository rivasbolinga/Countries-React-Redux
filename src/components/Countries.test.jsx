import { expect } from 'vitest';
import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import Countries from './Countries';

test('Countries component should match snapshot', () => {
  const countryData = {
    num: 1,
    flag: {
      png: 'flag.png',
      alt: 'Flag',
    },
    name: 'Country',
    capital: 'Capital',
    region: 'Continent',
  };
  const { container } = render(
    <Router>
      {/* eslint-disable-next-line react/jsx-props-no-spreading */}
      <Countries {...countryData} />
    </Router>,
  );
  expect(container).toMatchSnapshot();
});
