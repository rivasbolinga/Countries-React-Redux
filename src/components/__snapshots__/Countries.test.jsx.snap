// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Countries component should match snapshot 1`] = `
<div>
  <div
    class="sc-beqWaB bBCdOB country-card"
  >
    <a
      class="link"
      href="/country/1"
    >
      <img
        alt="Flag"
        class="flag"
        src="flag.png"
      />
      <div
        class="text"
      >
        <h2>
          Country
        </h2>
        <p>
          <span>
            Capital:
          </span>
           
          Capital
        </p>
        <p>
          <span>
            Continent:
          </span>
           
          Continent
        </p>
      </div>
    </a>
  </div>
</div>
`;
